### ISSUE TITLE
Can I use the automatic schema generator for non-referenced classes?

### ISSUE THREAD

**Issue by aronschmidt** [play-swagger/play-swagger#21]
Background is: We use jsonapi.org in our play project, which auto-generates its routes, so we don't have them in the routes file. Obviously I have to write the specs for that completely on my own, but at least the schemas could be auto generated. Maybe by another variable in the build.sbt file like:

```
swaggerAdditionalSchemas := Seq("jsonapiorg.resources.MyRestDTO")
```

I'd imagine it not too hard to be implemented. What do you think?

**Comment by Javakky-pxv** [comment]
Thank you for getting in touch. @aronschmidt 
Are you using [scala-jsonapi](https://github.com/scala-jsonapi/scala-jsonapi) ?
I don't have knowledge about it, so I would be grateful if you could provide comments on more specific usage for sample projects or proposed methods.

**Comment by aronschmidt** [comment]
No, we're using [qvantel/jsonapi-scala](https://github.com/qvantel/jsonapi-scala). But tbh we want to replace it by normal routes bc it adds much more overhead to our api compared to what we really use. I just wanted a little help for writing our openapi specs, bc I have to do that now and it will take some time to fully replace jsonapi.

Oh, I didn't see this repo is a fork of the iheartradio one. Maybe I should ask my question there?

**Comment by Javakky-pxv** [comment]
@aronschmidt 
Thank you for your reply.

> Oh, I didn't see this repo is a fork of the iheartradio one. Maybe I should ask my question there?
No, this repository is the master now. I have taken over from the iheartradio maintainer and am now developing it. I believe if you comment on the upstream repository, you might not receive a response.

I still don't fully understand the details of [qvantel/jsonapi-scala](https://github.com/qvantel/jsonapi-scala). I'd like to see a sample project using these in the Play Framework, or a more detailed explanation of the desired functionality.

So, you want to generate openapi json from jsonapi, is that correct?

**Comment by aronschmidt** [comment]
> So, you want to generate openapi json from jsonapi, is that correct?

That'd be a dream, yes. But for now I'd be happy if I could simply define a list of case class references, so the schemas for them get auto generated and inserted into the `swagger.json` file.

**Comment by aronschmidt** [comment]
I just figured out how to do what I wanted. When you reference the case class in my conf/swagger.yml like:

`$ref: '#/components/schemas/jsonapiorg.resources.MyRestDTO'`

Even though IntelliJ shows an "Unresolved reference" error in static file analysis, the referenced case classes schema will be generated by play-swagger and the resulting swagger.json is correct.
